### PR TITLE
systemc-cci: enable CMake builds

### PR DESCRIPTION
**Pull request by staskau**
Specify library name and version:  **systemc-cci/1.0.0**

Add CMakeLists.txt to enable cmake-based build, which among other things enables Windows build.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/systemc-cci/1.0.0/conanfile.py [hunk 1]
```diff
@@ -47,12 +45,12 @@ def build(self):
...
             tools.patch(**patch)
 
-        env_build = AutoToolsBuildEnvironment(self)
-        args = ['CONAN_MAKE_FILE={}'.format(
-            os.path.join(self.build_folder, "conanbuildinfo.mak"))]
-        with tools.chdir(os.path.join(self._source_subfolder, "src")):
-            env_build.make(args=args, target='clean')
-            env_build.make(args=args)
+        cmake = CMake(self, parallel=True)
+        cmake.verbose = True
```
Comment (madebr) [inline recipes/systemc-cci/1.0.0/conanfile.py:49]:
```suggestion
```

**Review by madebr — APPROVED** [review]

File: recipes/systemc-cci/1.0.0/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+set(LIBRARY_VERSION_MAJOR 1 CACHE STRING "major version" FORCE)
+set(LIBRARY_VERSION_MINOR 0 CACHE STRING "minor version" FORCE)
+set(LIBRARY_VERSION_PATCH 0 CACHE STRING "minor version" FORCE)
+set(LIBRARY_VERSION ${LIBRARY_VERSION_MAJOR}.${LIBRARY_VERSION_MINOR}.${LIBRARY_VERSION_PATCH} CACHE STRING "version" FORCE)
+
+set(LIBRARY_NAME cciapi)
+
+set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+set(POSITION_INDEPENDENT_CODE ON)
```
Comment (uilianries) [inline recipes/systemc-cci/1.0.0/CMakeLists.txt:14]:
```suggestion
```
It's automatically managed by fPIC option

File: recipes/systemc-cci/1.0.0/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.8)
+
+project(cci)
```
Comment (uilianries) [inline recipes/systemc-cci/1.0.0/CMakeLists.txt:3]:
```suggestion
project(systemc CXX)
```
enforce the language

File: recipes/systemc-cci/1.0.0/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+set(LIBRARY_VERSION ${LIBRARY_VERSION_MAJOR}.${LIBRARY_VERSION_MINOR}.${LIBRARY_VERSION_PATCH} CACHE STRING "version" FORCE)
+
+set(LIBRARY_NAME cciapi)
+
+set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+set(POSITION_INDEPENDENT_CODE ON)
+
+include(${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (uilianries) [inline recipes/systemc-cci/1.0.0/CMakeLists.txt:17]:
```suggestion
conan_basic_setup(KEEP_RPATHS)
```
better compatibility on Mac

File: recipes/systemc-cci/1.0.0/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	src/cci_cfg/cci_report_handler.cpp
+	src/cci_core/cci_name_gen.cpp
+	src/cci_core/cci_value_converter.cpp
+	src/cci_core/cci_value.cpp
+	src/cci_utils/broker.cpp
+	src/cci_utils/consuming_broker.cpp
+)
+
+add_library(${LIBRARY_NAME} ${SOURCES})	
+set_property(TARGET ${LIBRARY_NAME} PROPERTY POSITION_INDEPENDENT_CODE ON)
```
Comment (uilianries) [inline recipes/systemc-cci/1.0.0/CMakeLists.txt:45]:
```suggestion
```
automatically managed by fPIC option in the recipe

File: recipes/systemc-cci/1.0.0/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+target_include_directories(${LIBRARY_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/src)
+target_include_directories(${LIBRARY_NAME} PUBLIC ${RAPIDJSON}/include)
+
+install(TARGETS ${LIBRARY_NAME}
+  EXPORT ${PROJECT_NAME}Targets            # for downstream dependencies
```
Comment (uilianries) [inline recipes/systemc-cci/1.0.0/CMakeLists.txt:50]:
```suggestion
```
do not generated cmake files, it should be provided by conan

File: recipes/systemc-cci/1.0.0/CMakeLists.txt (hunk 4 above)
Comment (staskau) [reply recipes/systemc-cci/1.0.0/CMakeLists.txt:17]: Mac build is currently not supported

**Review by uilianries — CHANGES_REQUESTED** [review]
@staskau consider the follow PR as a fix and improvement. 

https://github.com/Minres/conan-center-index/pull/1

This recipe is outdated, so I updated some important parts.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**